### PR TITLE
Fix capitalizaton

### PR DESCRIPTION
**Pull request by jperk51**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

Change capitalization of strings for gender identity options

## Related issue(s)
department-of-veterans-affairs/va-mobile-app/issues/5351

## Testing done
Existing tests cover this case

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: lib/va_profile/models/gender_identity.rb [hunk 1]
```diff
@@ -13,8 +13,8 @@ class GenderIdentity < Base
       OPTIONS = {
         'M' => 'Man',
         'B' => 'Non-Binary',
```
Comment (TKDickson) [inline lib/va_profile/models/gender_identity.rb:15]: Per the design, also Non-binary - thanks!

File: modules/mobile/docs/schemas/GenderIdentityEdit.yml [hunk 2]
```diff
@@ -21,8 +21,8 @@ properties:
             example:
               M: "Man"
               B: "Non-Binary"
```
Comment (TKDickson) [inline modules/mobile/docs/schemas/GenderIdentityEdit.yml:23]: Per the design, also Non-binary - thanks!

**Review by TKDickson — APPROVED** [review]
Thanks for the non-binary fix as well

**Review by kpethtel — APPROVED** [review]
👍

**Review by ryan-mcneil — APPROVED** [review]
❤️

**Pull request merged**